front/app/containers/ProjectsShowPage/SucessModal.tsx

```diff
@@ -29,51 +32,31 @@ const SuccessModal = ({ projectId }: Props) => {
   const { data: phases } = usePhases(projectId);
 
   const [queryParams] = useSearchParams();
-  const showModalParam = queryParams.get('show_modal');
-  const phaseIdParam = queryParams.get('phase_id');
+  const [showModalParam] = useState<boolean>(!!queryParams.get('show_modal'));
```

Comment:
Could we avoid this local state and use the router for state management instead?

Reply:
Same mechanism- we want to capture the params from the URL and then remove them, because they are only passed to enable certain one-time behaviors that you don't want for random people that you later share the URL with. I didn't change this functionality, I just refactored it a bit